src/enzyme_ad/jax/Passes/CanonicalizeFor.cpp

```diff
@@ -689,14 +689,84 @@ struct WhileToForHelper {
   AddIOp addIOp;
   BlockArgument indVar;
   size_t afterArgIdx;
-  bool computeLegality(bool sizeCheck, Value lookThrough = nullptr) {
+
+  bool checkPredicate() {
+    auto cmpRhs = cmpIOp.getRhs();
```

Comment:
also cc @Pangoraw for context (this is supporting scf.while -> for for loops whose computation is in the before region), cc'd since you were looking at hlo related earlier